module/Makefile.bsd

```diff
@@ -423,25 +423,34 @@ CFLAGS.zil.c= -Wno-cast-qual
 CFLAGS.zio.c= -Wno-cast-qual
 CFLAGS.zrlock.c= -Wno-cast-qual
 CFLAGS.zfs_zstd.c= -Wno-cast-qual -Wno-pointer-arith
-CFLAGS.entropy_common.c= -fno-tree-vectorize -U__BMI__
-CFLAGS.error_private.c= -fno-tree-vectorize -U__BMI__
-CFLAGS.fse_decompress.c= -fno-tree-vectorize -U__BMI__
-CFLAGS.pool.c= -fno-tree-vectorize -U__BMI__
-CFLAGS.xxhash.c= -fno-tree-vectorize -U__BMI__
-CFLAGS.zstd_common.c= -fno-tree-vectorize -U__BMI__
-CFLAGS.fse_compress.c= -fno-tree-vectorize -U__BMI__
-CFLAGS.hist.c= -fno-tree-vectorize -U__BMI__
-CFLAGS.huf_compress.c= -fno-tree-vectorize -U__BMI__
-CFLAGS.zstd_compress.c= -fno-tree-vectorize -U__BMI__
-CFLAGS.zstd_compress_literals.c= -fno-tree-vectorize -U__BMI__
-CFLAGS.zstd_compress_sequences.c= -fno-tree-vectorize -U__BMI__
-CFLAGS.zstd_compress_superblock.c= -fno-tree-vectorize -U__BMI__
-CFLAGS.zstd_double_fast.c= -fno-tree-vectorize -U__BMI__
-CFLAGS.zstd_fast.c= -fno-tree-vectorize -U__BMI__
-CFLAGS.zstd_lazy.c= -fno-tree-vectorize -U__BMI__
-CFLAGS.zstd_ldm.c= -fno-tree-vectorize -U__BMI__
-CFLAGS.zstd_opt.c= -fno-tree-vectorize -U__BMI__
-CFLAGS.huf_decompress.c= -fno-tree-vectorize -U__BMI__
-CFLAGS.zstd_ddict.c= -fno-tree-vectorize -U__BMI__
-CFLAGS.zstd_decompress.c= -fno-tree-vectorize -U__BMI__
-CFLAGS.zstd_decompress_block.c= -fno-tree-vectorize -U__BMI__
+CFLAGS.entropy_common.c=-U__BMI__
+CFLAGS.error_private.c= -U__BMI__
+CFLAGS.fse_decompress.c= -U__BMI__
+CFLAGS.pool.c= -U__BMI__
+CFLAGS.xxhash.c= -U__BMI__
+CFLAGS.zstd_common.c= -U__BMI__
+CFLAGS.fse_compress.c= -U__BMI__
+CFLAGS.hist.c= -U__BMI__
+CFLAGS.huf_compress.c= -U__BMI__
+CFLAGS.zstd_compress.c= -U__BMI__
+CFLAGS.zstd_compress_literals.c= -U__BMI__
+CFLAGS.zstd_compress_sequences.c= -U__BMI__
+CFLAGS.zstd_compress_superblock.c= -U__BMI__
+CFLAGS.zstd_double_fast.c= -U__BMI__
+CFLAGS.zstd_fast.c= -U__BMI__
+CFLAGS.zstd_lazy.c= -U__BMI__
+CFLAGS.zstd_ldm.c= -U__BMI__
+CFLAGS.zstd_opt.c= -U__BMI__
+CFLAGS.huf_decompress.c= -U__BMI__
+CFLAGS.zstd_ddict.c= -U__BMI__
+CFLAGS.zstd_decompress.c= -U__BMI__
+CFLAGS.zstd_decompress_block.c= -U__BMI__
+
+# Woe betide ye who override this, for yours is the kingdom of mysterious segfaults
+# absolutely everywhere.
+#
+# Free warning to anyone considering it - -march=native or the like later in the line
+# will undo -mgeneral-regs-only, and gcc's -O2 starting in 12 does autovectorizing.
+#
+# Good luck.
+CFLAGS += -mgeneral-regs-only -fno-tree-vectorize
```

Comment:
I am not sure how far back FreeBSD's compiler support goes, but the same remark about `-mgeneral-regs-only` being a potential problem applies here.